db/bigtable.go

```diff
@@ -1186,9 +1186,18 @@ func (bigtable *Bigtable) GetValidatorBalanceStatistics(startEpoch, endEpoch uin
 		}
 
 		g.Go(func() error {
-
+			select {
+			case <-gCtx.Done():
+				return nil
+			default:
+			}
 			ranges := bigtable.getEpochRanges(fromEpoch, toEpoch)
 			err := bigtable.tableBeaconchain.ReadRows(ctx, ranges, func(r gcp_bigtable.Row) bool {
+				select {
+				case <-gCtx.Done():
+					return false
+				default:
+				}
```

Comment:
Shouldn't this be obsolete as the call above also has the context and should stop reading rows accordingly?

This would apply to more parts in this commit.

Reply:
removed